src/dispatching/dispatcher.rs

```diff
@@ -367,6 +392,45 @@ where
         }
     }
 
+    async fn gc_workers_if_needed(&mut self) {
+        if self.workers.len() <= self.gc_worker_count_trigger {
+            return;
+        }
+
+        self.gc_workers().await;
+    }
+
+    #[inline(never)]
+    async fn gc_workers(&mut self) {
```

Comment:
I suggest renaming this function to something more specific. `fn gc_workers` doesn't say much.

Also, what's the reason for `#[inline(never)]`?

Reply:
If you have a better naming idea I would be happy to listen to it :)

Maybe something like `remove_inactive_workers`?

> Also, what's the reason for #[inline(never)]?

We call it in "hot-ish" loop, but only if a rare condition is met. This is basically the way to mark function as cold.

Reply:
Yes, `remove_inactive_workers` would be better.

Reply:
> We call it in "hot-ish" loop, but only if a rare condition is met. This is basically the way to mark function as cold.

Let's add a comment then, I think it's useful information that this function isn't called frequently.